### ISSUE TITLE
Divider

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#107]
- Replace `---` with a horizontal rule (divider)
  - This can mainly be handled in `<Child>`
  - The `onChange` event in `<Editable>` should force a blur so the cursor thought is re-rendered. 
- When the cursor is on a divider, do not allow character insertions (only allow delete or navigate)

**Comment by thackley** [comment]
Because the cursor is not visible when it's on a divider, we need to highlight the divider. Also need to make sure that we can delete the divider.

**Comment by thackley** [comment]
@raineorshine I'm not sure how we want to handle "Do not merge thoughts when deleting divider". When thinking about it, it seems like we'd want to be able to backspace and convert the divider into two dashes but the way it is now, it does merge thoughts and become six dashes. Do we want to have a way to handle going from a divider to two dashes? Or completely delete the whole divider, taking back to just an empty thought?

**Comment by raineorshine** [comment]
Let's completely delete the divider and the thought that contains it!

**Comment by thackley** [comment]
@raineorshine I'm stumped when it comes to these two: 
- Also need to make sure that we can delete the divider.
- Allow navigation to select divider.

When we made the divider component, we didn't make it editable so I'm not too sure how to handle that. I tried a few different things but none of them seemed to work. Also, I'm able to delete dividers except a divider is the very first thought. 

Thoughts?
